src/usersInvites/usersInvites.service.ts

```diff
@@ -0,0 +1,110 @@
+import { HttpException, HttpStatus, Inject, Injectable } from '@nestjs/common';
+import { UserInvite } from '../mail/interface/userInvite.interface';
+import {
+  FITNESS_LOGGER_SERVICE,
+  FitnessLoggerService,
+} from '../logger/logger.service';
+import { MAIL_SERVICE, IMailService } from '../mail/mail.service';
+import { IUsersInvitesDAO, USERS_INVITES_DAO } from '../DAO/usersInvitesDAO';
+import { USERS_SERVICE, IUsersService } from '../users/users.service';
+import { UserInvites } from './interface/userInvites.interface';
+import { INVITES_SERVICE, IInvitesService } from '../invites/invites.service';
+import { inviteConfig } from '../config/mail/invite/invite.config';
+import { configApp } from '../config/configApp';
+import { SendMail } from '../mail/interface/sendMail.interface';
+
+export const USERS_INVITES_SERVICE = 'USERS INVITES SERVICE';
+
+export interface IUsersInvitesService {
+  createAndSendUserInvite(userInvite: UserInvite): Promise<UserInvites>;
+  sendInvite(mail: SendMail): Promise<UserInvites>;
+  createUserInvite(userInvite: UserInvites): Promise<UserInvites>;
+  updateUserInvite(userInvite: UserInvites): Promise<UserInvites>;
+}
+
+@Injectable()
+export class UsersInvitesService implements IUsersInvitesService {
+  constructor(
+    @Inject(MAIL_SERVICE) private mailService: IMailService,
+    @Inject(USERS_SERVICE) private readonly usersService: IUsersService,
+    @Inject(INVITES_SERVICE) private readonly invitesService: IInvitesService,
+    @Inject(USERS_INVITES_DAO) private usersInvitesDAO: IUsersInvitesDAO,
+    @Inject(FITNESS_LOGGER_SERVICE)
+    private readonly logger: FitnessLoggerService,
+  ) {
+    this.logger.setContext('UsersInvitesService');
+  }
+
+  async createAndSendUserInvite(userInvite: UserInvite): Promise<UserInvites> {
+    const user = await this.usersService.saveUser(userInvite);
+
+    const invite = await this.invitesService.saveInvite(inviteConfig.expiresAt);
+
+    await this.createUserInvite({ inviteId: invite.id, userId: user.id });
+
+    const dataInvite: SendMail = {
+      to: user.email,
+      from: inviteConfig.from,
+      subject: inviteConfig.subject,
+      template: inviteConfig.template,
+      url: await this.makeUrl(invite.invite),
+    };
```

Comment:
please, create a separate method to encapsulate `dataInvite` object creation. I would name it `makeInvitePayload`